### PR TITLE
3: Migrate to Now

### PR DESCRIPTION
**Pull request by mulholio**
This PR is a migration to a serverless paradigm and hosting on Now (issue #3 )

@WhiT3Kr0w This is working fine on my local machine, but the lambdas aren't working when I deploy by running `now`. The final thing to do here should be to add Now to the whitelist for RDS. We might be able to do this with an `"alias"` param in `now.json`

Some things to note:
- I have taken a couple of bits out of the code (e.g. sending 404s and other status codes) but these should all be added as separate issues.
- There are a few other minor but breaking changes:
  - Name API is now at `api/v1/foods?name=<name>`
  - Id of food is found at `api/v1/food/:id` to fit with REST conventions
  - Some changes to `api.js`

### CODE REVIEW

**Comment by mulholio** [comment]
@WhiT3Kr0w I will resolve merge conflicts

**Comment by mulholio** [comment]
Also, feel free to click the deployment link from the Now bot above. We should always get this preview for deploys from now on!

File: connect/testData.js [hunk 1]
```diff
@@ -1,13 +1,3 @@
...
-  foods: 'SELECT * FROM eatwell_fooddata;',
-  statusCheck:
-    'SELECT val FROM eatwell.status WHERE status.key = "status";',
-  foodById: id => `SELECT * FROM eatwell_fooddata WHERE id = "${id}";`,
-  foodByName: name =>
-    `SELECT * FROM eatwell_fooddata WHERE foodname LIKE "%${name}%";`,
-  recommendations: `SELECT rec.id, fd.foodname, fd2.foodname as "recommendation", cont.name FROM eatwell_recommendations rec, eatwell_fooddata fd, eatwell_fooddata fd2, eatwell_contributors cont WHERE rec.food_id = fd.id AND rec.foodrec_id=fd2.id AND rec.contributor_id=cont.id;`,
-};
-
 const testFoods = {
```
Comment (mulholio) [inline connect/testData.js]: I have left this in as it might be useful for actual testing later but atm it is unused.

File: connect/api.js [hunk 2]
```diff
@@ -1,50 +1,27 @@
...
     .trim()
     .toLowerCase();
 
-const fetchValues = endpoint => {
-  fetch(endpoint)
-    .then(response => response.json())
-    .then(data => data.value);
+const fetchValue = async endpoint => {
+  const res = await fetch(endpoint);
+  console.log('res', res);
```
Comment (WhiT3Kr0w) [inline connect/api.js]: Maybe worth cleaning up the console.logs?

**Comment by WhiT3Kr0w** [comment]
> Also, feel free to click the deployment link from the Now bot above. We should always get this preview for deploys from now on!

Looks great. Will wait for the conflicts' fix and will have another look.

**Comment by mulholio** [comment]
@WhiT3Kr0w all done for you

File: components/Recommendations.js [hunk 3]
```diff
@@ -12,7 +12,9 @@ const Recommendations = () => {
   const [recommendations, setRecommendations] = useState([]);
 
   const updateResults = () => {
-    fetchRecommendations().then(values => setRecommendations(values));
+    fetchRecommendations().then(({ recommendations }) =>
+      setRecommendations(recommendations)
```
Comment (WhiT3Kr0w) [inline components/Recommendations.js]: I had as values to not confuse with the recommendations state.
But we can leave this way if it doesn't cause issues.

**Comment by WhiT3Kr0w** [comment]
> This PR is a migration to a serverless paradigm and hosting on Now (issue #3 )
> 
> @WhiT3Kr0w This is working fine on my local machine, but the lambdas aren't working when I deploy by running `now`. The final thing to do here should be to add Now to the whitelist for RDS. We might be able to do this with an `"alias"` param in `now.json`
> 
> Some things to note:
> 
> * I have taken a couple of bits out of the code (e.g. sending 404s and other status codes) but these should all be added as separate issues.
> * There are a few other minor but breaking changes:
>   
>   * Name API is now at `api/v1/foods?name=<name>`
>   * Id of food is found at `api/v1/food/:id` to fit with REST conventions
>   * Some changes to `api.js`

Not sure how to make RDS visible to NOW serverless deployments. As currently I can only set an IP address in whitelist. Maybe we could setup some proxy.
I reckon we need to have the conversation about which DB to use and take this into account.

**Comment by mulholio** [comment]
> Not sure how to make RDS visible to NOW serverless deployments. As currently I can only set an IP address in whitelist. Maybe we could setup some proxy.
> I reckon we need to have the conversation about which DB to use and take this into account.

Now recommend using SSH and a password. Is this something you can achieve? https://github.com/zeit/now-cloud/issues/41

Can you allow read-only access from any IP to get this merged?

File: components/Recommendations.js (hunk 3 above)
Comment (mulholio) [reply components/Recommendations.js]: Yep, you're right. Still confusing, even if it works.

To resolve this, I've gotten rid of the destructuring and made 'fetchRecommendations' and other fetch functions return non-nested Recommendations etc which feels far more natural.

**Comment by WhiT3Kr0w** [comment]
> > Not sure how to make RDS visible to NOW serverless deployments. As currently I can only set an IP address in whitelist. Maybe we could setup some proxy.
> > I reckon we need to have the conversation about which DB to use and take this into account.
> 
> Now recommend using SSH and a password. Is this something you can achieve? [zeit/now-cloud#41](https://github.com/zeit/now-cloud/issues/41)
> 
> Can you allow read-only access from any IP to get this merged?

I don't think we can do this from RDS.
Worth having a look at MongoDB implementations that allow this.

I'm good to have this merged so we can continue the work.

**Pull request merged**